exports/security_group_inputs.tf

```diff
@@ -0,0 +1,194 @@
+  #    EOT
+
+  #  default     = true
+  #  description = <<-EOT
+  #    Set `true` to enable Terraform `create_before_destroy` behavior on the created security group.
+  #    We only recommend setting this `false` if you are upgrading this module and need to keep
+  #    the existing security group from being replaced.
+  #    Note that changing this value will always cause the security group to be replaced.
+  #    EOT
+}
+
+variable "security_group_create_timeout" {
+  type        = string
+  default     = "10m"
+  description = "How long to wait for the security group to be created."
+}
+
+variable "security_group_delete_timeout" {
+  type        = string
+  default     = "15m"
+  description = <<-EOT
+    How long to retry on `DependencyViolation` errors during security group deletion from
+    lingering ENIs left by certain AWS services such as Elastic Load Balancing.
+    EOT
+}
+
+#############################################################################################
+## Special note about inline_rules_enabled and revoke_rules_on_delete
+##
+## The security-group inputs inline_rules_enabled and revoke_rules_on_delete should not
+## be exposed in other modules unless there is a strong reason for them to be used.
+## We discourage the use of inline_rules_enabled and we rarely need or want
+## revoke_rules_on_delete, so we do not want to clutter our interface with those inputs.
+##
+## If someone wants to enable either of those options, they have the option
+## of creating a security group configured as they like
+## and passing it in as the target security group.
+#############################################################################################
+
+#
+#
+#### The variables below can be omitted if not needed, and may need their descriptions modified
+#
+#
+variable "vpc_id" {
+  type        = string
+  description = "The ID of the VPC where the Security Group will be created."
+}
+
+variable "revoke_security_group_rules_on_delete" {
+  type        = bool
+  default     = false
+  description = <<-EOF
+    Instruct Terraform to revoke all of the Security Group's attached ingress and egress rules before deleting
+    the security group itself. This is normally not needed.
+    EOF
+}
+
+variable "allow_all_egress" {
+  type        = bool
+  default     = true
+  description = <<-EOT
+    If `true`, the created security group will allow egress on all ports and protocols to all IP address.
+    If this is false and no egress rules are otherwise specified, then no egress will be allowed.
+    EOT
+}
+
+variable "additional_security_group_rules" {
+  type        = list(any)
+  default     = []
+  description = <<-EOT
+    A list of Security Group rule objects to add to the created security group, in addition to the ones
+    this module normally creates. (To suppress the module's rules, set `create_security_group` to false
+    and supply your own security group via `associated_security_group_ids`.)
```

Comment:
```suggestion
    and supply your own security group(s) via `associated_security_group_ids`.)
```

Reply:
I don't know what I was thinking. There is nothing for this module to do if it is not creating a security group and not adding rules to the target security group. Users should just set `enabled = false`. This needs a bigger fix.

Reply:
OIC, this is for the recommended inputs to consumers of this module. This is generally correct and I will implement this recommended change.